metrix-mapping/src/main/java/com/powsybl/metrix/mapping/log/RangeLogWithVariableChanged.java

```diff
@@ -0,0 +1,89 @@
+package com.powsybl.metrix.mapping.log;
+
+public class RangeLogWithVariableChanged extends AbstractLogBuilder implements LogDescriptionBuilder {
+
+    private String oldValue;
+    private String toVariable;
+    private String newValue;
+    private boolean disabled = false;
+    private String problemDescription;
+    private String actionDescription;
+    private String notIncludedVariable;
+    private String value;
+    private String id;
+    private String minValue;
+    private String maxValue;
+
+    protected static final String MAPPING_RANGE_PROBLEM = "mapping range problem" + LABEL_SEPARATOR;
+
+    protected static final String BC_RANGE_PROBLEM = "base case range problem" + LABEL_SEPARATOR;
+
+    public RangeLogWithVariableChanged oldValue(String oldValue) {
+        this.oldValue = oldValue;
+        return this;
+    }
+
+    public RangeLogWithVariableChanged newValue(double newValue) {
+        this.newValue = formatDouble(newValue);
+        return this;
+    }
+
+    public RangeLogWithVariableChanged toVariable(String toVariable) {
+        this.toVariable = toVariable;
+        return this;
+    }
+
+    public RangeLogWithVariableChanged disabled(boolean disabled) {
+        this.disabled = disabled;
+        return this;
+    }
+
+    public String getLabel(String problemDescription, String actionDescription) {
+        return problemDescription + oldValue + " changed to " + (toVariable.isEmpty() ? "0" : actionDescription) + toVariable;
```

Comment:
QCA : RMQ => what do you think of String.format( "{a: %s, b: %s, c: %s}", a, b, c ); to avoid this type of concatenation